File: Data Structures & Algorithms/sliding-window-maximum/submission-0.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        if k > len(nums):
            return [max(nums)] * len(nums)

        res = [-999999] * (len(nums) - k + 1) 

        l = 0
        r = k - 1

        

        while r < len(nums):
            res[l] = max(nums[l:r + 1])
            l += 1
            r += 1

        return res

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] result = new int[nums.length];
        Arrays.fill(result, 1);
        int current = 1;
        for (int i = 0; i < nums.length; i++) {
            result[i] *= current;
            current *= nums[i];
        }
        current = 1;
        for (int i = nums.length - 1; i >= 0; i--) {
            result[i] *= current;
            current *= nums[i];
        }
        return result;
    }
}  
